### ISSUE TITLE
Reservation create EJS view

### ISSUE THREAD

**Issue by haietza** [JaysGitLab/cs5667-final-project-revdev#18]
EJS view for form to create a new reservation

**Comment by haietza** [comment]
@patrickbeekman I have the route/controller/view changes pushed up and I'm not getting the error message anymore. However, the createRes view never loads when I run the application - it just keeps trying to load and then times out. I can't figure out why the page isn't loading - there are no error messages to reference. If you want to pull the updated branch and take a look, maybe together we can figure it out?

**Comment by haietza** [comment]
@patrickbeekman I got it! It was a silly typo, of course. I'm working on pulling the eventTypes from the db into the view for the select list on the form. I'll push and comment here when that is up so you can pull and write the tests. Sorry for the delay!

**Comment by patrickbeekman** [comment]
Wow you are truly amazing! No worries about the delay! That sounds good I can start looking into the edit and delete stuff and attempt to implement that tonight!

**Comment by haietza** [comment]
@patrickbeekman Whew, okay I finally have the reservation create stuff working. There are a couple other things I think we need to add on this, but not sure if they belong here or as additional issues. We can talk about it more tomorrow in class. Did you want to take this issue over from here and add the tests, or would you like me to do that for the create reservation stuff and you can keep going with the list, etc stuff?

**Comment by patrickbeekman** [comment]
@haietza Awesome job! Yeah we can definitely talk more tomorrow about the extra issues you had. I can take a look at writing the tests for the routes and controllers but I am still very shaky on it. Whatever I don't finish with that we can look at tomorrow! I started working on the list stuff last night but couldn't get the page to load so I'm sitting down now to do some debugging right now!

**Comment by haietza** [comment]
@patrickbeekman If it's okay, I'm going to start working on the tests for the createRes view and routes/controller methods implemented so far. So I can be productive in class this morning ;).

**Comment by haietza** [comment]
I'm having a hard time getting the tests to pass as a suite. They all pass when run individually, but throw different errors each time when run together. I'm still working on it.

The TODOs I mentioned us needing to refer back to are:
* The ref to the Event document in the Reservation model does not actually store all the details of the Event in the Reservation document. It only stores the ObjectID. I looked at using populate() to get the Event data into the Reservation document when it is created but that did not appear to do anything. So we'll need to revisit this in order to make sure the Reservation documents are accurate.
* We need to add a 'success' message when a Reservation is successfully added. I tried implementing this in the current code, but it does not display when redirected to the home page, so we'll need to revisit.
* As we discussed, we'll need to revisit the datetime HTML form field usage. Currently, this does not work in Firefox or IE - it just displays a plain text field. We could add instructions for users of those browsers, or try to implement separate date and time fields that will work in all browsers.
